src/librustc_mir/interpret/eval_context.rs

```diff
@@ -500,7 +624,9 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M
         &mut self,
         rvalue: &mir::Rvalue<'tcx>,
         place: &mir::Place<'tcx>,
-    ) -> EvalResult<'tcx> {
+    ) -> EvalResult<'tcx>
+        where M: Clone + Eq + Hash,
+    {
```

Comment:
Maybe add this to `Machine` itself. Eg:

```
trait Machine: Clone + Eq + Hash {
[...]
```

Reply:
That seems fine, `Machine` isn't really used for anything but this.